### PR TITLE
[java][jersey2] Differentiate request with no body vs request that contains the null value

### PR DESCRIPTION
**Pull request by sebastien-rosset**
Two scenarios need to be differentiated:
1. HTTP requests may have an empty body, e.g. HTTP GET requests.
1. HTTP requests may also have a body with the 'null' value.

We need to be able to differentiate these two cases when serializing the body. In particular, for HTTP GET request which have an empty body, the input object was set to null, and it was serialized as the 'null' value instead of the empty string.

This leads to incorrect message digest being calculated.

### PR checklist

- [x] Read the [contribution guidelines](https://github.com/openapitools/openapi-generator/blob/master/CONTRIBUTING.md).
- [x] If contributing template-only or documentation-only changes which will change sample output, [build the project](https://github.com/OpenAPITools/openapi-generator#14---build-projects) beforehand.
- [x] Run the shell script `./bin/generate-samples.sh`to update all Petstore samples related to your fix. This is important, as CI jobs will verify _all_ generator outputs of your HEAD commit as it would merge with master. These must match the expectations made by your contribution. You may regenerate an individual generator by passing the relevant config(s) as an argument to the script, for example `./bin/generate-samples.sh bin/config/java*`. For Windows users, please run the script in [Git BASH](https://gitforwindows.org/).
- [x] File the PR against the [correct branch](https://github.com/OpenAPITools/openapi-generator/wiki/Git-Branches): `master`
- [x] Copy the [technical committee](https://github.com/openapitools/openapi-generator/#62---openapi-generator-technical-committee) to review the pull request if your PR is targeting a particular programming language.

### CODE REVIEW

File: samples/client/petstore/java/jersey2-java8/src/main/java/org/openapitools/client/ApiClient.java [hunk 1]
```diff
@@ -787,7 +794,7 @@ public String escapeString(String str) {
    * @return String
    * @throws ApiException API exception
    */
-  public String serializeToString(Object obj, Map<String, Object> formParams, String contentType) throws ApiException {
+  public String serializeToString(BodyHolder obj, Map<String, Object> formParams, String contentType) throws ApiException {
```
Comment (sebastien-rosset) [inline samples/client/petstore/java/jersey2-java8/src/main/java/org/openapitools/client/ApiClient.java:797]: @wing328 , I'm a bit confused why `updateParamsForAuth` invokes serializeToString, but the request body is serialized using a different function. That means an authorization scheme that processes the message body may provide incorrect authorization data.
For example, the message digest may be different.

**Comment by sebastien-rosset** [comment]
@bbdouglas (2017/07) @sreeshas (2017/08) @jfiala (2017/08) @lukoyanov (2017/09) @cbornet (2017/09) @jeff9finger (2018/01) @karismann (2019/03) @Zomzog (2019/04) @lwlee2608 (2019/10) @bkabrda (2020/01)

File: samples/client/petstore/java/jersey2-java8/src/main/java/org/openapitools/client/ApiClient.java (hunk 1 above)
Comment (wing328) [reply samples/client/petstore/java/jersey2-java8/src/main/java/org/openapitools/client/ApiClient.java:790]: I agree ideally they should be using the same function to come up with the request body.

Comment (sebastien-rosset) [reply samples/client/petstore/java/jersey2-java8/src/main/java/org/openapitools/client/ApiClient.java:790]: thanks. I propose that we tackle this in a follow-up PR, what do you think? I see some discrepancies between serializeToString and serialize.

**Pull request closed without merging**